examples/eth_bots/execute_agent_trades.py

```diff
@@ -219,38 +220,60 @@ async def async_match_contract_call_to_trade(
 
     """
     # TODO: figure out fees paid
+    # TODO: clean up this function, DRY it up to reduce number of statements
+    # pylint: disable=too-many-statements
+    trade_amount: int = trade.trade.trade_amount.scaled_value
+    max_deposit: int = trade_amount
+
     # TODO: The following variables are hard coded for now, but should be specified in the trade spec
-    # HyperdriveMarketAction does have min_amount_out
-    trade_amount: int = trade_object.trade.trade_amount.scaled_value
-    max_deposit = trade_amount
-    min_output = 0
     min_apr = int(1)
     max_apr = int(1e18)
     as_underlying = True
-    match trade_object.trade.action_type:
+    match trade.trade.action_type:
         case MarketActionType.OPEN_LONG:
+            min_output = 0
+            fn_args = (trade_amount, min_output, account.checksum_address, as_underlying)
+            preview_result = smart_contract_preview_transaction(hyperdrive_contract, account, "openLong", *fn_args)
+            min_output = (
+                FixedPoint(scaled_value=preview_result["bondProceeds"])
+                * (FixedPoint(1) - trade.trade.slippage_tolerance)
+            ).scaled_value
             fn_args = (trade_amount, min_output, account.checksum_address, as_underlying)
+
             trade_result = await async_transact_and_parse_logs(
                 web3,
                 hyperdrive_contract,
                 account,
                 "openLong",
                 *fn_args,
             )
-            maturity_time_years = FixedPoint(trade_result.maturity_time) / elftime.TimeUnit.YEARS.value
-            mint_time_years = maturity_time_years - hyperdrive_market.position_duration.years
+            maturity_time_seconds = trade_result.maturity_time_seconds
             wallet_deltas = WalletDeltas(
                 balance=Quantity(
                     amount=-trade_result.base_amount,
                     unit=TokenType.BASE,
                 ),
-                longs={FixedPoint(mint_time_years): Long(trade_result.bond_amount)},
+                longs={FixedPoint(maturity_time_seconds): Long(trade_result.bond_amount)},
             )
         case MarketActionType.CLOSE_LONG:
-            mint_time_years: FixedPoint = trade_object.trade.mint_time
-            decoded_maturity_time_years = mint_time_years + hyperdrive_market.position_duration.years
-            decoded_maturity_time = int(decoded_maturity_time_years * elftime.TimeUnit.YEARS.value)
-            fn_args = (decoded_maturity_time, trade_amount, min_output, account.checksum_address, as_underlying)
+            maturity_time_seconds = trade.trade.mint_time
```

Comment:
we should rename this from mint time if we're not going to store mint time in it